modules/shared/src/main/scala/io/constellationnetwork/security/Hash.scala

```diff
@@ -25,11 +24,19 @@ object hash {
 
   object Hash {
 
-    def hashCodeFromBytes(bytes: Array[Byte]): HashCode =
-      Hashing.sha256().hashBytes(bytes)
-
-    def fromBytes(bytes: Array[Byte]): Hash =
-      Hash(hashCodeFromBytes(bytes).toString)
+    def sha256FromBytes(bytes: Array[Byte]): Array[Byte] = {
+      val md = MessageDigest.getInstance("SHA-256")
+      md.update(bytes)
+      md.digest()
+    }
+
+    def fromBytes(bytes: Array[Byte]): Hash = {
+      val sha256Bytes = sha256FromBytes(bytes)
+      val sha256String = sha256Bytes
+        .foldLeft(new StringBuilder(64))((acc, x) => acc.append(s"%02x".format(x)))
+        .toString
+      Hash(sha256String)
+    }
```

Comment:
Have you verified if new methods produce same hashes as `Hashing` one?

Reply:
I would be super super careful doing this. Hashing is the base of the whole codebase. We should test it again starting from genesis, revalidating the chain (partially at least) to make sure we won't make a breaking change and the hashing remains consistent.

Reply:
See https://github.com/effects-ai/tessellation/blob/sha256_tests/modules/shared/src/test/scala/io/constellationnetwork/security/GuavaJSATest.scala

Reply:
What's the output?

Reply:
All green, since all tests use the same java.security.MessageDigest inside, plus

```
MessageDigest.clone() 32389 ms
MessageDigest.getInstance() 33180 ms
Guava 32336 ms
```

YMMV, though.